pkg/cover/backend/elf.go

```diff
@@ -422,34 +416,39 @@ func readCoverPoints(file *elf.File, info *symbolInfo, arch string) ([2][]uint64
 	if err != nil {
 		return pcs, fmt.Errorf("failed to read .text: %v", err)
 	}
-	for i := 0; i < len(data); i++ {
-		if arch == targets.AMD64 {
-			if i > len(data)-callLen {
-				break
-			}
-		} else if arch == targets.ARM64 {
-			if i < callLen-1 {
+	amd64 := true
```

Comment:
I think prefer to use targets.AMD64? we still need to port other arch like ARM32 in future.

Reply:
I wanted to remove the string comparison from the inner loop (may be slow).
OK, if you want to add arm32, let's go full data-driven.
I've updated the PR, please take another look.

Reply:
looks nice and clean, thanks